diff --git a/src/wl/sys/wl_linux.c b/src/wl/sys/wl_linux.c
--- a/src/wl/sys/wl_linux.c	2021-04-15 13:57:31.209657503 +0800
+++ b/src/wl/sys/wl_linux.c	2021-04-15 13:57:10.071108458 +0800
@@ -805,8 +805,8 @@
 	pci_set_master(pdev);
 
 	pci_read_config_dword(pdev, 0x40, &val);
-	if ((val & 0x0000ff00) != 0)
-		pci_write_config_dword(pdev, 0x40, val & 0xffff00ff);
+	if ((val & 0x0000ff00) != 0) {
+		pci_write_config_dword(pdev, 0x40, val & 0xffff00ff);}
 		bar1_size = pci_resource_len(pdev, 2);
 		bar1_addr = (uchar *)ioremap_nocache(pci_resource_start(pdev, 2),
 			bar1_size);
