diff --git a/Makefile b/Makefile
index 90d1a2a1fdcd4..3ae3e80d3e958 100644
--- a/Makefile
+++ b/Makefile
@@ -753,7 +753,7 @@ stackp-flags-$(CONFIG_STACKPROTECTOR_STRONG)      := -fstack-protector-strong
 
 KBUILD_CFLAGS += $(stackp-flags-y)
 
-ifdef CONFIG_CC_IS_CLANG
+ifndef CONFIG_CC_IS_CLANG
 KBUILD_CPPFLAGS += -Qunused-arguments
 KBUILD_CFLAGS += -Wno-format-invalid-specifier
 KBUILD_CFLAGS += -Wno-gnu
@@ -768,7 +768,9 @@ else
 # Warn about unmarked fall-throughs in switch statement.
 # Disabled for clang while comment to attribute conversion happens and
 # https://github.com/ClangBuiltLinux/linux/issues/636 is discussed.
-KBUILD_CFLAGS += $(call cc-option,-Wimplicit-fallthrough,)
+#KBUILD_CFLAGS += $(call cc-option,-Wimplicit-fallthrough,)
+KBUILD_CFLAGS += -Wno-gnu-variable-sized-type-not-at-end
+KBUILD_CFLAGS += -Wno-implicit-fallthrough
 endif
 
 # These warnings generated too much noise in a regular build.
diff --git a/Makefile b/Makefile
index 33f62d84c625f..f8146becf3910 100644
--- a/Makefile
+++ b/Makefile
@@ -1073,7 +1073,7 @@ mod_sign_cmd = true
 endif
 export mod_sign_cmd
 
-HOST_LIBELF_LIBS = $(shell pkg-config libelf --libs 2>/dev/null || echo -lelf)
+HOST_LIBELF_LIBS = $(shell $(BUILD_PKG_CONFIG) libelf --libs 2>/dev/null || echo -lelf)
 
 has_libelf = $(call try-run,\
                echo "int main() {}" | $(HOSTCC) -xc -o /dev/null $(HOST_LIBELF_LIBS) -,1,0)
