diff --git a/src/fydeos.com/power_daemon/backlight_manager/backlight_manager.go b/src/fydeos.com/power_daemon/backlight_manager/backlight_manager.go
index ca705e9..736172a 100644
--- a/src/fydeos.com/power_daemon/backlight_manager/backlight_manager.go
+++ b/src/fydeos.com/power_daemon/backlight_manager/backlight_manager.go
@@ -10,7 +10,7 @@ import (
   "github.com/godbus/dbus/v5"
   "strconv"
   "io/ioutil"
-  pmpb "chromiumos/system_api/power_manager_proto"
+  pmpb "go.chromium.org/chromiumos/system_api/power_manager_proto"
   "fydeos.com/power_daemon/dbusutil"
 )
 
diff --git a/src/fydeos.com/power_daemon/suspend_manager/suspend_manager.go b/src/fydeos.com/power_daemon/suspend_manager/suspend_manager.go
index 260b4ba..a543d75 100644
--- a/src/fydeos.com/power_daemon/suspend_manager/suspend_manager.go
+++ b/src/fydeos.com/power_daemon/suspend_manager/suspend_manager.go
@@ -8,7 +8,7 @@ import (
   "log"
   "errors"
   "github.com/godbus/dbus/v5"
-  pmpb "chromiumos/system_api/power_manager_proto"
+  pmpb "go.chromium.org/chromiumos/system_api/power_manager_proto"
   "fydeos.com/power_daemon/dbusutil"
 )
 
